Sources/GithubAPI.swift

```diff
@@ -0,0 +1,97 @@
+/*
+ Copyright 2018 Google LLC
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ https://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+import Foundation
+import PerfectCURL
+import PerfectHTTP
+import PerfectLogger
+import PerfectCrypto
+
+public class GithubAPI {
+  static let githubBaseURL = "https://api.github.com"
+  static let retryCount = 0
+
+  class func addLabelsToIssue(url: String, labels: [String]) {
+    let labelsURL = url + "/labels"
+    do {
+      let request = CURLRequest(labelsURL, .postString(labels.description))
+      addAPIHeaders(to: request)
+      let response = try request.perform()
+      if GithubAuth.refreshCredentialsIfUnauthorized(response: response) {
+        addLabelsToIssue(url: url, labels: labels)
```

Comment:
Is this recursing?

Reply:
yes but only if it works to get an authorized response for getting a refresh token and getting access again to the api, otherwise it wont recurse.

Reply:
it is supposed to recurse only once basically